extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/discovery/AstroDiscoveryService.java

```diff
@@ -31,39 +36,70 @@
 public class AstroDiscoveryService extends AbstractDiscoveryService {
     private final Logger logger = LoggerFactory.getLogger(AstroDiscoveryService.class);
     private static final int DISCOVER_TIMEOUT_SECONDS = 30;
+    private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
     private LocationProvider locationProvider;
+    private final ExpressionThreadPoolExecutor scheduledExecutor;
+    private Runnable backgroundJob;
+
+    private static ThingUID SUN_THING = new ThingUID(THING_TYPE_SUN, LOCAL);
+    private static ThingUID MOON_THING = new ThingUID(THING_TYPE_MOON, LOCAL);
 
     /**
-     * Creates a AstroDiscoveryService with disabled autostart.
+     * Creates a AstroDiscoveryService with enabled autostart.
      */
     public AstroDiscoveryService() {
-        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, false);
+        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, true);
+
+        scheduledExecutor = ExpressionThreadPoolManager.getExpressionScheduledPool("astro");
     }
 
     @Override
     protected void startScan() {
         logger.debug("Starting Astro discovery scan");
-
         PointType location = locationProvider.getLocation();
-
         if (location == null) {
             logger.debug("LocationProvider.getLocation() is not set -> Will not provide any discovery results");
             return;
         }
+        createResults(location);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        Expression expression = null;
+        try {
+            expression = new CronExpression(
```

Comment:
I doubt that this is what he meant.
Every discovery service already HAS a `scheduler` available. I see no reason why this should not be used here. Check the docs at https://www.eclipse.org/smarthome/documentation/development/bindings/discovery-services.html#background-discovery (and please improve them, if they weren't clear enough on that point!).